src/ctia/task/migrate_es_stores.clj

```diff
@@ -0,0 +1,206 @@
+(ns ctia.task.migrate-es-stores
+  (:require [clj-momo.lib.es
+             [document :as es-doc]
+             [index :as es-index]]
+            [clojure.string :as string]
+            [clojure.tools.logging :as log]
+            [ctia
+             [init :refer [init-store-service! log-properties]]
+             [properties :as p]
+             [store :refer [stores]]]
+            [ctia.task.migrations :refer [available-migrations]]))
+
+(def default-batch-size 100)
+
+(defn compose-migrations
+  "compose migrations from a list of keywords"
+  [migration-keys]
+  (let [migrations
+        (vals (select-keys available-migrations
+                           migration-keys))]
+    (if (seq migrations)
+      (apply comp migrations)
+      (do (log/warn "target migration not found, copying data")
+          (map identity)))))
+
+(defn prefixed-index [index prefix]
+  (let [version-trimmed (string/replace index #"^v\w+.\w+.\w_" "")]
```

Comment:
I think this regex is slightly too risky. Why not: `#"^v[^_]*_"` to remove everything up to the first underscore that shouldn't be part of any semantic version number.

In your regex the `.` are considered as "any char" and not as dot.